src/cli/cli_tcat.cpp

```diff
@@ -108,6 +111,66 @@ static void HandleBleSecureReceive(otInstance               *aInstance,
     IgnoreReturnValue(otBleSecureFlush(aInstance));
 }
 
+template <> otError Tcat::Process<Cmd("vendorid")>(Arg aArgs[])
+{
+    otError                  error            = OT_ERROR_NONE;
+    static const char *const kVendorIdTypes[] = {"empty", "oui24", "oui36", "discriminator", "ianapen"};
+
+    if (aArgs[0].IsEmpty())
+    {
+        if (sVendorInfo.mDeviceId != nullptr)
+        {
+            OutputLine("type %s, value: %s", kVendorIdTypes[sVendorInfo.mDeviceIdType], sVendorInfo.mDeviceId);
+        }
+        else
+        {
+            OutputLine("%s", kVendorIdTypes[OT_TCAT_DEVICE_ID_EMPTY]);
+        }
+        ExitNow();
+    }
+
+    if (aArgs[0] == kVendorIdTypes[OT_TCAT_DEVICE_ID_OUI24])
+    {
+        sVendorInfo.mDeviceIdType = OT_TCAT_DEVICE_ID_OUI24;
+    }
+    else if (aArgs[0] == kVendorIdTypes[OT_TCAT_DEVICE_ID_OUI36])
+    {
+        sVendorInfo.mDeviceIdType = OT_TCAT_DEVICE_ID_OUI36;
+    }
+    else if (aArgs[0] == kVendorIdTypes[OT_TCAT_DEVICE_ID_DISCRIMINATOR])
+    {
+        sVendorInfo.mDeviceIdType = OT_TCAT_DEVICE_ID_DISCRIMINATOR;
+    }
+    else if (aArgs[0] == kVendorIdTypes[OT_TCAT_DEVICE_ID_IANAPEN])
+    {
+        sVendorInfo.mDeviceIdType = OT_TCAT_DEVICE_ID_IANAPEN;
+    }
+    else if (aArgs[0] == kVendorIdTypes[OT_TCAT_DEVICE_ID_EMPTY])
+    {
+        sVendorInfo.mDeviceIdType = OT_TCAT_DEVICE_ID_EMPTY;
+        sVendorInfo.mDeviceId     = nullptr;
+        ExitNow();
+    }
+    else
+    {
+        ExitNow(error = OT_ERROR_INVALID_ARGS);
+    }
+
+    if (aArgs[1].GetLength() < (OT_TCAT_MAX_VENDORID_SIZE) && !aArgs[1].IsEmpty())
+    {
+        strcpy(sVendorId, aArgs[1].GetCString());
```

Comment:
for all these ID types, the user would typically like to provide a hex number to set the bits of the ID to the correct OUI number, or IANA PEN, or discriminator. In OT there is already a function for this. So e.g. if I type:

```
tcat vendorid ianapen e378
```

Then the ID sequence is set to 2 bytes 0xe3, 0x78, or IANA PEN 58232.

I tried the draft code below and that seems to work. 

```
    if (aArgs[1].GetLength() <= (2*(OT_TCAT_MAX_VENDORID_SIZE-1)) && !aArgs[1].IsEmpty())
    {
        uint16_t length = OT_TCAT_MAX_VENDORID_SIZE;
        error = aArgs[1].ParseAsHexString(length, sVendorId);
        VerifyOrExit(error == OT_ERROR_NONE);
        sVendorId[length] = 0; // null-terminator
        LogDebg("parsed into: %u %u %u %u %u %u len=%u", sVendorId[0],sVendorId[1],sVendorId[2],sVendorId[3],sVendorId[4],sVendorId[5],length);
        sVendorInfo.mDeviceId = (char *) sVendorId;
    }
```

Reply:
Fixed, also added some tests for it.